include/url.h

```diff
@@ -79,6 +79,11 @@ namespace Url
          */
         Url& abspath();
 
+        /**
+         * Evaluate 'other' relative to this URL, placing the result in this object.
+         */
+        Url& relative(const std::string& other);
```

Comment:
This library has two purposes -- both to parse the piece and to normalize them. As such, I think mutating makes sense to perform a series of transformations and then reassemble the pieces. I think that makes more sense than making copies.

To avoid disruption but also take advantage of reusing the the `base` URL, I propose a new API, `relative_to` -- it is clearer about the relationship between the object and `other` and would accept `const std::string& other` or `const Url& other`. How does that sound?